### ISSUE TITLE
`sst init` fails on js-cloudflare template

### ISSUE THREAD

**Issue by jack-davies** [sst/ion#190]
There seems to be an issue with hujson patching when following the Cloudflare [quickstart](https://ion.sst.dev/docs/start/cloudflare/):

```
$ mkdir my-worker && cd my-worker
$ npm init -y
$ sst init --verbose
...
✓  Template: js
✓  Using: cloudflare
...
time=2024-04-07T00:35:52.288+11:00 level=INFO msg=patching file=package.json
time=2024-04-07T00:35:52.288+11:00 level=ERROR msg="exited with error" err="hujson: patch operation 0: value not found"
```

The first patch operation succeeds if I add `"dependencies": {}` to `package.json`:

```diff
{
  "name": "my-worker",
  "version": "1.0.0",
  "description": "",
  "main": "index.js",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "keywords": [],
  "author": "",
  "license": "ISC",
+ "dependencies": {}
}
```

But this results in a new error:

```
time=2024-04-07T00:43:49.308+11:00 level=INFO msg=patching file=package.json
time=2024-04-07T00:43:49.308+11:00 level=ERROR msg="exited with error" err="hujson: patch operation 1: value not found"
```

In this case, additionally adding

```diff
+ "devDependencies": {}
```

Unfortunately does not resolve this error, due to what appears to be an unquoted `/` character in the `@cloudflare/workers-types` package name, which hujson interprets as a path in [https://github.com/sst/ion/commit/8f7f5b32d732b87c0102f9f00997d61477ea600e#diff-5a4c4fbffb0fa1b66641a818c3f99ee9af45593e0cabb174c1c2366f7bf1e28aR18](https://github.com/sst/ion/commit/8f7f5b32d732b87c0102f9f00997d61477ea600e#diff-5a4c4fbffb0fa1b66641a818c3f99ee9af45593e0cabb174c1c2366f7bf1e28aR18).

Indeed, adding the malformed

```diff
+ "devDependencies": {"@cloudflare": {}}
```

Results in a successful run with the following patch produced:

```
  "dependencies": {"sst":"^3.0.4"},
  "devDependencies": {"@cloudflare": {"workers-types":"^3.0.4"}}
```

**Comment by ruchernchong** [comment]
I am also facing this problem.

**Comment by garysassano** [comment]
Same exact issue.

**Comment by garysassano** [comment]
> ❯ sst init --verbose
time=2024-04-07T14:34:15.712+02:00 level=INFO msg="checking for pulumi" path=/home/user/.config/sst/bin/pulumi
time=2024-04-07T14:34:15.834+02:00 level=INFO msg="checking for bun" path=/home/user/.config/sst/bin/bun
×  SST project already exists

**Comment by garysassano** [comment]
```sh
❯ sst init --verbose
time=2024-04-07T14:34:15.712+02:00 level=INFO msg="checking for pulumi" path=/home/user/.config/sst/bin/pulumi
time=2024-04-07T14:34:15.834+02:00 level=INFO msg="checking for bun" path=/home/user/.config/sst/bin/bun
×  SST project already exists
```

**Comment by nick-0101** [comment]
Getting the same thing

**Comment by ruchernchong** [comment]
> ```shell
> ❯ sst init --verbose
> time=2024-04-07T14:34:15.712+02:00 level=INFO msg="checking for pulumi" path=/home/user/.config/sst/bin/pulumi
> time=2024-04-07T14:34:15.834+02:00 level=INFO msg="checking for bun" path=/home/user/.config/sst/bin/bun
> ×  SST project already exists
> ```

Remove `sst.config.ts` and run the command again.

**Comment by garysassano** [comment]
@ruchernchong It's the same result, just with some extra errors for the first run.

```sh
❯ sst init --verbose
time=2024-04-08T03:56:09.448+02:00 level=INFO msg="checking for pulumi" path=/home/user/.config/sst/bin/pulumi
time=2024-04-08T03:56:09.572+02:00 level=INFO msg="checking for bun" path=/home/user/.config/sst/bin/bun

   ███████╗███████╗████████╗
   ██╔════╝██╔════╝╚══██╔══╝
   ███████╗███████╗   ██║   
   ╚════██║╚════██║   ██║   
   ███████║███████║   ██║   
   ╚══════╝╚══════╝   ╚═╝   

>  JS project detected. This will...
   - use the JS template
   - create an sst.config.ts

✓  Template: js

✓  Using: cloudflare

time=2024-04-08T03:56:13.448+02:00 level=INFO msg="creating project" name=my-worker
time=2024-04-08T03:56:13.448+02:00 level=INFO msg="copying template" path=templates/js-cloudflare/files/sst.config.ts
time=2024-04-08T03:56:13.448+02:00 level=INFO msg="copying template" path=templates/js-cloudflare/files/tsconfig.json
time=2024-04-08T03:56:13.449+02:00 level=INFO msg=patching file=package.json
time=2024-04-08T03:56:13.449+02:00 level=ERROR msg="exited with error" err="hujson: patch operation 0: value not found"
×  Unexpected error occurred. Please check the logs or run with --verbose for more details.
```

**Comment by ruchernchong** [comment]
> @ruchernchong It's the same result, just with some extra errors for the first run.
> 
> 
> 
> ```sh
> 
> ❯ sst init --verbose
> 
> time=2024-04-08T03:56:09.448+02:00 level=INFO msg="checking for pulumi" path=/home/user/.config/sst/bin/pulumi
> 
> time=2024-04-08T03:56:09.572+02:00 level=INFO msg="checking for bun" path=/home/user/.config/sst/bin/bun
> 
> 
> 
>    ███████╗███████╗████████╗
> 
>    ██╔════╝██╔════╝╚══██╔══╝
> 
>    ███████╗███████╗   ██║   
> 
>    ╚════██║╚════██║   ██║   
> 
>    ███████║███████║   ██║   
> 
>    ╚══════╝╚══════╝   ╚═╝   
> 
> 
> 
> >  JS project detected. This will...
> 
>    - use the JS template
> 
>    - create an sst.config.ts
> 
> 
> 
> ✓  Template: js
> 
> 
> 
> ✓  Using: cloudflare
> 
> 
> 
> time=2024-04-08T03:56:13.448+02:00 level=INFO msg="creating project" name=my-worker
> 
> time=2024-04-08T03:56:13.448+02:00 level=INFO msg="copying template" path=templates/js-cloudflare/files/sst.config.ts
> 
> time=2024-04-08T03:56:13.448+02:00 level=INFO msg="copying template" path=templates/js-cloudflare/files/tsconfig.json
> 
> time=2024-04-08T03:56:13.449+02:00 level=INFO msg=patching file=package.json
> 
> time=2024-04-08T03:56:13.449+02:00 level=ERROR msg="exited with error" err="hujson: patch operation 0: value not found"
> 
> ×  Unexpected error occurred. Please check the logs or run with --verbose for more details.
> 
> ```

Oh you are running the Cloudflare provider. It is not working properly currently.

**Comment by thdxr** [comment]
this should be fixed in v0.0.280
